### PR TITLE
[Merge_into_R1] Fix for EOS-18435 - object listing timeout issue

### PR DESCRIPTION
**Pull request by DPG17**
The fix is basically cherry-picked from 'main' branch @ commit id 2789e03b73095b37b348f48d02d0d18d7a692b99

### CODE REVIEW

**Review by knrajnambiar76 — APPROVED** [review]

File: ut/s3_get_bucket_action_test.cc [hunk 1]
```diff
@@ -409,15 +415,28 @@ TEST_F(S3GetBucketActionTest, GetNextObjectsSuccessfulDelimiterLastKey) {
...
+  EXPECT_CALL(*(motr_kvs_reader_factory->mock_motr_kvs_reader),
+              next_keyval(_, _, _, _, _, _))
+      .Times(AtLeast(1))
+      .WillRepeatedly(InvokeWithoutArgs(
+           this, &S3GetBucketActionTest::call_next_keyval_successful));
+  bucket_meta_factory->mock_bucket_metadata->set_objects_version_list_index_oid(
+      object_list_indx_oid);
+  bucket_meta_factory->mock_bucket_metadata->set_object_list_index_oid(
+      object_list_indx_oid);
+  //  EXPECT_CALL(*bucket_meta_factory->mock_bucket_metadata,
```
[the remark this answers is not in the document]
Comment (DPG17) [reply ut/s3_get_bucket_action_test.cc]: done.

**Comment by DPG17** [comment]
Pre merge jenkins job:
 http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/278  (in progress)

**Review by evgeniy-brazhnikov — APPROVED** [review]

**Comment by DPG17** [comment]
Triggered pre merge build on the correct jenkins pipeline:
 - http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-cortx-1.0/9/  (build is in progress)

**Comment by vimalraghwani20** [comment]
Code is merged in to 1.0 branch

**Pull request merged**